bcipy/signal/model/pca_rda_kde/pca_rda_kde.py

```diff
@@ -0,0 +1,126 @@
+import pickle
+from pathlib import Path
+from typing import List
+
+import numpy as np
+from bcipy.signal.model.base_model import ModelEvaluationReport, SignalModel
+from bcipy.signal.model.pca_rda_kde.classifier import RegularizedDiscriminantAnalysis
+from bcipy.signal.model.pca_rda_kde.cross_validation import cost_cross_validation_auc, cross_validation
+from bcipy.signal.model.pca_rda_kde.density_estimation import KernelDensityEstimate
+from bcipy.signal.model.pca_rda_kde.dimensionality_reduction import ChannelWisePrincipalComponentAnalysis
+from bcipy.signal.model.pca_rda_kde.pipeline import Pipeline
+from bcipy.signal.exceptions import SignalException
+
+
+class PcaRdaKdeModel(SignalModel):
+    def __init__(self, k_folds=10):
+        self.k_folds = k_folds
+        self.model = None
+        self._ready_to_predict = False
+
+    def fit(self, train_data: np.array, train_labels: np.array) -> SignalModel:
+        """
+        Train on provided data using K-fold cross validation and return self.
+
+        Parameters:
+            train_data: shape (Channels, Trials, Trial_length) preprocessed data
+            train_labels: shape (Trials,) binary labels
+
+        Returns:
+            trained likelihood model
+        """
+        model = Pipeline([ChannelWisePrincipalComponentAnalysis(var_tol=1e-5, num_ch=train_data.shape[0]),
+                          RegularizedDiscriminantAnalysis()])
+
+        # Find the optimal gamma + lambda values
+        arg_cv = cross_validation(train_data, train_labels, model=model, k_folds=self.k_folds)
+
+        # Get the AUC using those optimized gamma + lambda
+        model.pipeline[1].lam = arg_cv[0]
+        model.pipeline[1].gam = arg_cv[1]
+        _, sc_cv, y_cv = cost_cross_validation_auc(
+            model, 1, train_data, train_labels, arg_cv, k_folds=self.k_folds, split="uniform"
```

Comment:
Same here.. what does one represent?

Reply:
Looks like it's the index for which element in the `Pipeline` is being optimized. In fact I don't think this parameter gets used, and it must always be a pipeline with PCA -> RDA, where the thing being optimized is RDA (index `1`). 

I'll add to a ticket to do some cleanup/minimization of this code as well, because it could probably be shortened without changing the logic.